lib/validate.py

```diff
@@ -67,47 +94,83 @@ def box_plot_for_degree(log_directory, output_directory, channel_name):
+        print "Top 10 HUBS for Month [HITS]", next_month_dt.month, ":", top_hubs_m2
         print "Number of common HUBS (from 10) between above 2 months:", len(list(set(top_hubs_m1).intersection(top_hubs_m2)))
 
-        print "Top 10 Experts by keywords for Months", month, ":", top_keyword_overlap_m1
-        print "Top 10 Experts by keywords for Months", month + 1, ":", top_keyword_overlap_with_score_m2
+        print "Top 10 Experts by keywords for Months", dt.month, ":", top_keyword_overlap_m1
+        print "Top 10 Experts by keywords for Months", next_month_dt.month, ":", top_keyword_overlap_with_score_m2
         print "Number of common Experts by keywords (from 10) between above 2 months:", len(list(set(top_keyword_overlap_m1).intersection(top_keyword_overlap_with_score_m2)))
 
-        print "Top 10 AUTH for Month [HITS]", month, ":", top_auth_m1
-        print "Top 10 AUTH for Month [HITS]", month + 1, ":", top_auth_m2
+        print "Top 10 AUTH for Month [HITS]", dt.month, ":", top_auth_m1
+        print "Top 10 AUTH for Month [HITS]", next_month_dt.month, ":", top_auth_m2
         print "Number of common AUTH (from 10) between above 2 months:", len(list(set(top_auth_m1).intersection(top_auth_m2)))
         
-        print "Number of users common btw HUBS from HITS and Experts by Keywords (from 10) for month", month, ":",  len(list(set(top_keyword_overlap_m1).intersection(top_hubs_m1))) 
-        print "Number of users common btw AUTH from HITS and Experts by Keywords (from 10) for month", month, ":",  len(list(set(top_keyword_overlap_m1).intersection(top_auth_m1)))
-        print "Number of users common btw HUBS from HITS and AUTH from HITS (from 10) for month", month, ":",  len(list(set(top_hubs_m1).intersection(top_auth_m1)))
-        print "Number of users common btw HUBS, HITS and KEYWORDS", month, ":", len(set(list(set(top_keyword_overlap_m1).intersection(top_hubs_m1))).intersection(top_auth_m1))
+        print "Number of users common btw HUBS from HITS and Experts by Keywords (from 10) for month", dt.month, ":",  len(list(set(top_keyword_overlap_m1).intersection(top_hubs_m1)))
+        print "Number of users common btw AUTH from HITS and Experts by Keywords (from 10) for month", dt.month, ":",  len(list(set(top_keyword_overlap_m1).intersection(top_auth_m1)))
+        print "Number of users common btw HUBS from HITS and AUTH from HITS (from 10) for month", dt.month, ":",  len(list(set(top_hubs_m1).intersection(top_auth_m1)))
+        print "Number of users common btw HUBS, HITS and KEYWORDS", dt.month, ":", len(set(list(set(top_keyword_overlap_m1).intersection(top_hubs_m1))).intersection(top_auth_m1))
+
+
+def codelengths(log_directory, output_directory, channel_name, start_date, end_date):
+    """
+        The function iterate through the months in the given date range and computes the infomap number. It then plots a
+        box plot for the infomap numbers of the given month.
```

Comment:
box plot for the Infomap codelength numbers generated for all the whole months in the given time period.

Reply:
done